addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java

```diff
@@ -63,21 +69,26 @@ public String getLabel(Locale locale) {
     }
 
     @Override
-    public void process(AudioStream audioStream) throws UnsupportedAudioFormatException {
+    public void process(AudioStream audioStream)
+            throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
         AudioFormat format = audioStream.getFormat();
         if (!AudioFormat.WAV.isCompatible(format) && !AudioFormat.MP3.isCompatible(format)) {
             throw new UnsupportedAudioFormatException("Currently only MP3 and WAV formats are supported: ", format);
         }
 
         String url;
-        if (audioStream instanceof FixedLengthAudioStream) {
+        if (audioStream instanceof URLAudioStream) {
+            url = ((URLAudioStream) audioStream).getURL();
+        } else if (audioStream instanceof FixedLengthAudioStream) {
             // Since Squeezebox will make multiple requests for the stream, set a timeout on the stream
             url = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, STREAM_TIMEOUT).toString();
 
             if (AudioFormat.WAV.isCompatible(format)) {
-                url += ".wav";
+                // TODO add FileAudioStream.EXTENSION_SEPARATOR
```

Comment:
For the record: `FileAudioStream.WAV_EXTENSION` currently contains the value `".wav"` which is not a preferred way to go. Pointing to this [discussion](https://github.com/eclipse/smarthome/pull/3598#discussion_r120611406). I am working on an improvement for that in ESH and will adept it afterwards.